### PR TITLE
Reorganize documentation

### PR DESCRIPTION
**Pull request by melissawm**
# Description
This PR implements a reorganization of the documentation pages previously discussed with the team.

I have a number of notes below, but this PR implements the following:
- Reorganization
- Some formatting fixes, including removing trailing whitespace and converting document and section titles to `Sentence case`
- Updates to links, particularly code links that predated the move to `src/MaxText`.

This PR **does not** implement content changes.

# Tests

Documentation builds successfully locally.

# Checklist

Before submitting this PR, please make sure (put X in square brackets):
- [x] I have performed a self-review of my code. For an optional AI review, add the `gemini-review` label.
- [x] I have necessary comments in my code, particularly in hard-to-understand areas.
- [x] I have run end-to-end tests tests and provided workload links above if applicable.
- [x] I have made or will make corresponding changes to the doc if needed.

### CODE REVIEW

File: docs/guides/run_maxtext_localhost.md [hunk 1]
```diff
@@ -21,28 +21,26 @@ MaxText uses a primary YAML file, `configs/base.yml`, to manage its settings. Th
...
 ## Development
 Local development on a single host TPU/GPU VM is a convenient way to run MaxText on a single host. It doesn't scale to multiple hosts but is a good way to learn about MaxText. The following describes how to run Maxtext on TPU/GPU VMs.
 
-### Run Maxtext on Single Host VM
+### Run MaxText on single host VM
 1.  Create and SSH to the single host VM of your choice. You can use any available single host TPU, such as `v5litepod-8`, `v5p-8`, or `v4-8`. For GPUs, you can use `nvidia-h100-mega-80gb`, `nvidia-h200-141gb`, or `nvidia-b200`. For setting up a TPU VM, use the Cloud TPU documentation available at https://cloud.google.com/tpu/docs/managing-tpus-tpu-vm. For a GPU setup, refer to the guide at https://cloud.google.com/compute/docs/gpus/create-vm-with-gpus. 
 
 2.  Clone MaxText onto that VM.
     ```bash
-    # Clone the repository
```
Comment (melissawm) [inline docs/guides/run_maxtext_localhost.md:35]: Redundant comment removed.

File: docs/guides/run_maxtext_localhost.md [hunk 2]
```diff
@@ -21,28 +21,26 @@ MaxText uses a primary YAML file, `configs/base.yml`, to manage its settings. Th
...
     cd maxtext
     ```
 
 3.  Once you have cloned the repository, you have two primary options for setting up the necessary dependencies on your VM: Installing in a Python Environment, or building a Docker container. For single host workloads, we recommend to install dependencies in a python environment, and for multihost workloads we recommend the containerized approach.
 
-Within the root directory of the cloned repo, install dependencies and the pre-commit hook by running:
+Within the root directory of the cloned repo, create a virtual environment and install dependencies and the pre-commit hook by running:
 
 ```bash
-# Create a virtual environment and install dependencies
```
Comment (melissawm) [inline docs/guides/run_maxtext_localhost.md:45]: Redundant comment removed.

File: docs/guides/run_maxtext_via_pathways.md [hunk 3]
```diff
@@ -15,7 +15,7 @@
 -->
 
 (run-pathways)=
-# Guide: Running MaxText via Pathways
```
Comment (melissawm) [inline docs/guides/run_maxtext_via_pathways.md:18]: For consistency.

File: docs/guides/data_input_pipeline.md [hunk 4]
```diff
@@ -21,9 +21,9 @@ Currently MaxText has three data input pipelines:
 
 | Pipeline | Dataset formats | Features | Limitations |
 | -------- | --------------- | -------- | ----------- |
-| **[Grain](data_input_grain.md)** (recommanded)| [ArrayRecord](https://github.com/google/array_record) (random access, available through [Tensorflow Datasets](https://www.tensorflow.org/datasets/catalog/overview), or [conversion](https://github.com/google/array_record/tree/main/beam))<br>[Parquet](https://arrow.apache.org/docs/python/parquet.html) (sequential access) | With arrayrecord: fully deterministic, resilient to preemption; global shuffle <br>With parquet: performant; fully deterministic, resilient to preemption; hierarchical shuffle |  |
```
Comment (melissawm) [inline docs/guides/data_input_pipeline.md]: This typo and a number of other changes here were implemented in https://github.com/AI-Hypercomputer/maxtext/pull/2266 but never merged.

File: docs/reference/config_base.md [hunk 5]
```diff
@@ -0,0 +1,6 @@
+# Base configuration file
+
+MaxText uses a base configuration file to set default values for various parameters. This file is located at [`src/MaxText/configs/base.yml`](https://github.com/AI-Hypercomputer/maxtext/blob/main/src/MaxText/configs/base.yml) in the MaxText repository, and is shown below.
+
+```{literalinclude} ../../src/MaxText/configs/base.yml
```
Comment (melissawm) [inline docs/reference/config_base.md:5]: This creates a nice page with the full contents of base.yml. If you don't think this is useful, I can remove.

File: docs/tutorials/full_finetuning.md [hunk 6]
```diff
@@ -54,7 +54,7 @@ The conversion scripts do not use accelerators but need large host memory to per
 - For large size model (e.g. 70B model), this script requires large memory VM.
 - The script load and save weights in a single pass.
 
-### Sample Full Fine tuning Script
+### Sample full fine tuning script
```
Comment (melissawm) [inline docs/tutorials/full_finetuning.md:57]: Should we consistently say either "fine tuning" or "finetuning"? Both appear in the documentation in different documents.

File: docs/conf.py [hunk 7]
```diff
@@ -31,6 +31,7 @@
 extensions = [
     "myst_nb",
     "sphinx_design",
+    "sphinx_copybutton",
```
Comment (melissawm) [inline docs/conf.py:39]: This extension adds a button to every code block so it's instantly copied to the readers clipboard when clicked.

**Comment by melissawm** [comment]
Hi all, just rebased so it's up to date with `main`. Let me know if you have any feedback. Thanks!

File: docs/tutorials/sft_on_multi_host.md [hunk 8]
```diff
@@ -14,25 +14,25 @@
 # limitations under the License.
  -->
 
-# Supervised Fine-Tuning (SFT) with Deepseek-V3 Model
+# Supervised Fine-Tuning (SFT) with Deepseek-V3 model
```
Comment (jacoguzo) [inline docs/tutorials/sft_on_multi_host.md:17]: The naming here is a little confusing, it doesn't convey the progression from the Try SFT. Maybe we can name the file Try SFT with DeepSeek-V3.

File: docs/tutorials/sft.md [hunk 9]
```diff
@@ -14,10 +14,10 @@
  limitations under the License.
  -->
 
-# Try SFT!
+# Try SFT
```
Comment (jacoguzo) [inline docs/tutorials/sft.md:17]: Do we want to keep this consistent and change the name to Try SFT with Llama3 8B?

File: docs/tutorials.md [hunk 10]
```diff
@@ -41,7 +41,6 @@ You can also find other examples in the [MaxText repository](https://github.com/
 tutorials/first_run.md
 tutorials/pretraining.md
 tutorials/full_finetuning.md
-tutorials/run_llama2.md
 tutorials/grpo.md
```
Comment (jacoguzo) [inline docs/tutorials.md:44]: Should we change the order of these so that we have the SFT tutorials grouped first and then the GRPO ones next?

**Review by jacoguzo — APPROVED** [review]
